Added /spectator command

TheWadeSmith opened the pull request:
Interesting command. I don't see how people can break the rules with it. It will also help admins get around easier.

Commit:
Create Command_spectate

Commit:
Rename Command_spectate to Command_spectate.java

Wild1145 commented:
Trolling would be my only concern. Personally I would limit it to admins only and have this as an admin tool.

CrafterSmith12 commented:
How would they troll with it exactly?

JeromSar commented:
@CrafterSmith12 I don't understand why players would want to be in spectator mode. Could you enlighten me on that? :)

AlexFreedomMod commented:
Ops can grief with it I suggest making it for admins only kinda like a better invis potion

CrafterSmith12 commented:
@JeromSar It's manly for admins, however, I don't see an issue with players having it. Please correct me of I'm wrong.

JeromSar commented:
If I'm not mistaken, players in spectator mode are invisible to other players. Thus, I think that adding this command will only add chaos, and wouldn't benefit TF's community.

Wild1145 commented:
@JeromSar You are invisible to all players not in Spectator mode. Thus why I would suggest it be strictly for admins only, and not OP's, as while in spectator mode you cannot interact with the world or other players, it would be a very spy orientated tool and could be used to troll other players.

Commit:
Admin only

-Removed the feature to change other's gamemode..
-Removed the feature to change everyone's gamemode [/gmsp -a].
-Made it admin only (due to request).
-Removed console compatibility.

Commit:
Little fix

CrafterSmith12 commented:
@JeromSar

JeromSar commented on src/me/StevenLawson/TotalFreedomMod/Commands/Command_spectator.java:
```diff
@@ -0,0 +1,35 @@
+        }
+        else
+        {
+            if (!(senderIsConsole || TFM_AdminList.isSuperAdmin(sender)))
```
Only supers can execute this command, this is redundant.

Wild1145 commented:
@JeromSar Re-Open?

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/TotalFreedom/TotalFreedomMod?pullRequest=816) <br/>You should sign our Contributor License Agreement in order to get your pull request merged.<br/>

DUFCLiam commented:
Should be closed due to the release of 5.0; @JeromSar
